test/integration/test_seq_copy_int.py

```diff
@@ -139,6 +142,37 @@ def test_seq_copy(train_params: str,
             dev_source_factor_paths = [data['validation_source']]
             test_source_factor_paths = [data['test_source']]
 
+        # When using constrained decoding, rewrite the source file. Generating a mixture of
+        # sentences with and without constraints here is critical, since this can happen in production
+        # and also introduces sometimes some unanticipated interactions.
+        test_source_path = data['test_source']
+        test_target_path = data['test_target']
+        if constrained_decoding:
```

Comment:
couldn't this functionality be added to the `tmp_digits_dataset` function in common.py?